admin/server/users.go

```diff
@@ -284,6 +284,76 @@ func (s *Server) GetUser(ctx context.Context, req *adminv1.GetUserRequest) (*adm
 	return &adminv1.GetUserResponse{User: userToPB(user)}, nil
 }
 
+func (s *Server) DeleteUser(ctx context.Context, req *adminv1.DeleteUserRequest) (*adminv1.DeleteUserResponse, error) {
+	observability.AddRequestAttributes(
+		ctx,
+		attribute.String("args.email", req.Email),
+		attribute.String("args.org", req.Organization),
+	)
+
+	org, err := s.admin.DB.FindOrganizationByName(ctx, req.Organization)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	role, err := s.admin.DB.FindOrganizationRole(ctx, database.OrganizationRoleNameAdmin)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	currentUser, err := s.GetCurrentUser(ctx, &adminv1.GetCurrentUserRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	claims := auth.GetClaims(ctx)
+
+	isCurrentUser := currentUser.User.Email == req.Email
+	isSuperuser := claims.Superuser(ctx)
+
+	if !isCurrentUser && !isSuperuser {
+		return nil, status.Error(codes.PermissionDenied, "only superusers can delete other users")
+	}
```

Comment:
1. The `s.GetCurrentUser(...)` function call means one gRPC RPC handler calls another without using a real gPRC client. Even though it's probably safe here, we generally avoid that as it can mess with instrumentation and lead to edge cases where the correct middleware is not applied. Instead we prefer to pull shared logic between handlers into shared/utility functions or just duplicating the logic if it's just a few lines of code.
2. In this case, you can avoid the call by moving up the `s.admin.DB.FindUserByEmail(ctx, req.Email)` call later in the function and doing a check like `claims.OwnerType() == auth.OwnerTypeUser && claims.OwnerID() == user.ID`